src/modules/calculator/dtos/result.dto.ts

```diff
@@ -1,5 +1,11 @@
-export class ResultDto {
-  walletAddress: string;
-  cRecys: number;
-  wasteFootprint: number;
-}
+import { z } from 'zod';
+
+export const ResultDtoSchema = z.object({
+  walletAddress: z.string({ message: 'walletAddress must be a string' }),
```

Comment:
Could you update here to validate EVM wallet ?

```
walletAddress: z
    .string()
    .regex(/^0x[a-fA-F0-9]{40}$/, 'Invalid EVM wallet address format')
    .or(z.literal(''))
    .optional(),
```

https://github.com/detrash/recy-api/blob/staging/src/modules/recycling-report/dtos/create-recycling-report.dto.ts